pkg/user/tx_client.go

```diff
@@ -628,19 +663,41 @@ func (client *TxClient) ConfirmTx(ctx context.Context, txHash string) (*TxRespon
 				client.deleteFromTxTracker(txHash)
 				return nil, executionErr
 			}
+			// if it took more than 7 seconds to confirm, log it
+			if elapsed > 7*time.Second {
```

Comment:
we don't need this for traces, just add the event itself bc we'll see the time it took for it to get confirmed eventually. We would need smth like this more for metrics but we can discuss that later.